futures-sink/src/channel_impls.rs

```diff
@@ -1,62 +1,66 @@
-use {Async, Sink, AsyncSink, StartSend, Poll};
+use {Async, Sink, Poll};
 use futures_core::task;
-use futures_channel::mpsc::{Sender, SendError, UnboundedSender};
-
-fn res_to_async_sink<T>(res: Result<(), T>) -> AsyncSink<T> {
-    match res {
-        Ok(()) => AsyncSink::Ready,
-        Err(x) => AsyncSink::Pending(x),
-    }
-}
+use futures_channel::mpsc::{Sender, ChannelClosed, UnboundedSender};
 
 impl<T> Sink for Sender<T> {
     type SinkItem = T;
-    type SinkError = SendError<T>;
+    type SinkError = ChannelClosed<T>;
 
-    fn start_send(&mut self, cx: &mut task::Context, msg: T) -> StartSend<T, SendError<T>> {
-        self.start_send(cx, msg).map(res_to_async_sink)
+    fn poll_ready(&mut self, cx: &mut task::Context) -> Poll<(), Self::SinkError> {
+        self.poll_ready(cx)
     }
 
-    fn flush(&mut self, _: &mut task::Context) -> Poll<(), SendError<T>> {
-        Ok(Async::Ready(()))
+    fn start_send(&mut self, msg: T) -> Result<(), Self::SinkError> {
+        self.start_send(msg)
     }
 
-    fn close(&mut self, _: &mut task::Context) -> Poll<(), SendError<T>> {
+    fn start_close(&mut self) -> Result<(), Self::SinkError> {
+        Ok(())
```

Comment:
This has confused me in the past. I would expect to be able to try to close a channel from the sender side, even if I have multiple senders. Should this instead try to close the channel, instead of being a noop?

Reply:
Can you open an issue for this? I'd prefer to keep this PR scoped to just refactoring changes, not behavioral ones.